packages/transit-vehicle-overlay/src/WithCaret.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { FC, HTMLAttributes } from "react";
+import styled from "styled-components";
+
+import { InnerCaret, OuterCaret } from "./styled";
+import { VehicleComponentProps } from "./types";
+
+interface CaretOptions {
+  height?: number;
+  offset?: number;
+  position?: "inner" | "outer";
+  width?: number;
+}
+
+type IconContainerProps = HTMLAttributes<HTMLDivElement> &
+  VehicleComponentProps;
+
+/**
+ * Adds a caret to a component.
+ */
+export default function withCaret(
+  Component: FC<IconContainerProps>,
+  options?: CaretOptions
+): FC<IconContainerProps> {
+  const isInner = options?.position === "inner";
+  const height = options?.height || 0;
+  const width = options?.width || height * 2;
+  const halfWidth = width / 2;
+  const offset = options?.offset || 0;
+  const RawCaret = isInner ? InnerCaret : OuterCaret;
+  const SizedCaret = height
+    ? styled(RawCaret)`
+        &::before {
+          border-bottom-width: ${Math.max(0, height)}px;
+          border-left-width: ${halfWidth}px;
+          border-right-width: ${halfWidth}px;
+          margin-left: -${halfWidth}px;
+          top: -${offset + (isInner ? 0 : height)}px;
```

Comment:
I think this might be causing some issues with custom sized outer carets -- with inner carets things work fine. <img width="250" alt="Screen Shot 2022-11-07 at 8 41 42 AM" src="https://user-images.githubusercontent.com/86619099/200324899-3ef4e933-4dd0-40d9-9b35-95844667e5b9.png">

Reply:
Correct, the implementer has to choose an offset and size that look good together.